src/config/ttlConfig.ts

```diff
@@ -45,18 +45,27 @@ export class TTLConfig<T extends TTLConfig.Options, P extends JsonMap> extends C
 
   protected async init(): Promise<void> {
     const contents = await this.read(this.options.throwOnNotFound);
-    const purged = {} as TTLConfig.Contents<P>;
-    const date = new Date().getTime();
-    for (const [key, opts] of Object.entries(contents)) {
-      if (!this.isExpired(date, opts)) purged[key] = opts;
+    if (contents && this.hasExpiredEntries(contents)) {
+      this.setContents(this.removeExpiredEntries(contents));
+      await this.write();
+    } else {
+      this.setContents(contents);
```

Comment:
The previous code was excluding expired cache entries from the in-memory representation of the cache.

This change  removes the expired entries from in-memory representation AND saves the remaining keys back to cache file.

It is possible to have a cache file grow with expired entries, if the only way to remove the entry is by the code using the cache, via unset.

This change will keep the cache devoid of expired entries.

Reply:
configFile has problems with concurrency, and now init is calling write, which I think is asking for cross-save trouble.

What if we kept the `removeExpiredEntries` but attached it to a new `write` method override (so that it removes the expired bits and then calls super.write() or however y'all OOP folks would do it).  Then writes only write right when writes were already happening, right?